testsuite/cluster-test/src/main.rs

```diff
@@ -738,16 +742,24 @@ impl ClusterTestRunner {
         Ok(())
     }
 
-    fn run_health_check(&mut self) {
+    fn run_health_check(&mut self, duration: Duration) -> Result<()> {
+        let health_check_deadline = Instant::now() + duration;
         loop {
             let deadline = Instant::now() + Duration::from_secs(1);
             // Receive all events that arrived to aws log tail within next 1 second
             // This assumes so far that event propagation time is << 1s, this need to be refined
             // in future to account for actual event propagation delay
             let events = self.logs.recv_all_until_deadline(deadline);
-            let _ignore =
-                self.health_check_runner
-                    .run(&events, &HashSet::new(), PrintFailures::All);
+            let result = self
+                .health_check_runner
+                .run(&events, &HashSet::new(), PrintFailures::All);
+            let now = Instant::now();
+            if now > health_check_deadline {
+                break match result {
```

Comment:
You can simplify this to `break result.map(|_| ())`

Reply:
Also I think return here will be more obvious/appropriate then break